### ISSUE TITLE
Getting the output probabilities of the categories

### ISSUE THREAD

**Issue by Piyush0291** [jpmml/jpmml-xgboost#6]
Hi,

I am trying to get the raw predicted values of the categories(binary xgboost model in my case) from the PMML that is generated from the utility that you have shared. But I am only able to get the final predicted output from it. Is there a way I can get the output probabilities in the OutputField?

Thanks,

Piyush

**Comment by vruusmann** [comment]
All classification-type XGBoost models should be encoding a probability distribution.

Please provide the following information:
1) What is your JPMML-XGBoost version? If it's older than `1.0.6`, then please upgrade.
2) What is the exact name of your XGBoost objective function? Is it `binary:logistic`, or something else? Description such as "binary xgboost model" is not that helpful to me.

**Comment by vruusmann** [comment]
If you are using classification-type `binary:logistic` or `multi:softmax` (aka `multi:softprob`) objective functions, then the PMML representation of the XGBoost model contains an `Output` element, which encodes the probability distribution in the form of a list of `OutputField` elements. This is proven by `LogisticClassificationAudit` and `SoftMaxClassificationIris` test cases.

If you are using regression-type objective function, such as `reg:logistic`, then the `Output` element is not generated.

Closing this issue as invalid.

**Comment by Piyush0291** [comment]
Hi,

I am using binary:logistic as the objective function. I am just downloading the zip file containing the src folder and so I hope it is the latest. To re-iterate, I am building the model that has two labels(1 and 0). From the PMML file, I want to output the Probability_0 and Probability_1. Right now I am only getting the final classification i.e. 0 or 1

**Comment by vruusmann** [comment]
Exporting the `LogisticClassificationAudit` test case:
```
java -jar target/converter-executable-1.1-SNAPSHOT.jar --model-input src/test/resources/xgboost/LogisticClassificationAudit.model --fmap-input src/test/resources/csv/Audit.fmap --pmml-output LogisticClassificationAudit.pmml
```

Open the resulting file `LogisticClassificationAudit.pmml` in text editor and see lines 2907 -- 2910:
```
<Output>
	<OutputField name="probability_0" feature="probability" value="0"/>
	<OutputField name="probability_1" feature="probability" value="1"/>
</Output>
```
